### PR TITLE
OpenMP Parallelization and cover calculation tuning options.

### PR DESCRIPTION
**Pull request by michaelleerilee**
Added several cover calculation options controlled through command line. Added OpenMP parallelization to STARE index calculation. Achieves decent speedup and usage. May require setting -DCMAKE_EXE_LINKER_FLAGS=-lomp on cmake command line. Probably breaks autotools build. Cover calculation isn't a huge part of the execution time. Once built with OpenMP, one can use 'export OMP_NUM_THREADS=<int>' to trigger parallel execution. Yields about a factor of 5 speedup with 8 threads. Looks like MOD05 code is most mature.

### CODE REVIEW

**Comment by michaelleerilee** [comment]
Adding G-RING calculation here...

**Comment by captainkirk99** [comment]
Do you want to merge these changes? They look great.

After this is in I will add the equivalent stuff in the autotools build to find OpenMP...

**Comment by captainkirk99** [comment]
OK, part of the problem is that I believe you are using a different indentation for the code.

Looks like you are using 2. I will switch to that.

Are you using emacs? If so, send me your C++ format settings and I will use them.

**Comment by michaelleerilee** [comment]
Oh, I understand. Didn't think through what that indent-region would do re: git. I am using emacs, default settings. If you know of a good way to extract those settings, let me know and I'll run it.

**Comment by captainkirk99** [comment]
Do you have a .emacs file in your home directory? If so, look in it for anything related to C++ or CXX...

**Comment by captainkirk99** [comment]
I have to take the bike into the shop. ;-) I'll pick this up tomorrow morning or later today...

**Comment by michaelleerilee** [comment]
Can't find anything relevant in .emacs or .emacs.d/*

File: src/Modis09L2GeoFile.cpp [hunk 1]
```diff
@@ -118,24 +121,28 @@ Modis09L2GeoFile::readFile(const std::string fileName, int verbose, int quiet,
 						   sizeof(unsigned long long))))
 	return SSC_ENOMEM;
 
-    int level = 5;
+    int level = 27;
     STARE index(level, build_level);
```
Comment (captainkirk99) [inline src/Modis09L2GeoFile.cpp:125]: I believe, with your changes, that this line can be removed, which should improve performance...

**Comment by captainkirk99** [comment]
I have merged these code changes via a series of PRs. I will close this PR.

We still could use some testing of these new features, and I need to see what happens without openmp.

**Pull request closed without merging**